ci_framework/playbooks/06-deploy-edpm.yml

```diff
@@ -49,10 +49,46 @@
             name: libvirt_manager
             tasks_from: deploy_edpm_compute.yml
 
+        - name: Prepare for HCI deploy phase 1
+          when: cifmw_edpm_deploy_hci | default(false) | bool
+          ansible.builtin.import_role:
+            name: hci_prepare
+            tasks_from: phase1.yml
+
         - name: Deploy EDPM
           ansible.builtin.import_role:
             name: edpm_deploy
 
+- name: Clear edpm hosts facts to force refreshing in HCI deployments
+  hosts: edpm
+  tasks:
+    - name: Clear edpm hosts facts
+      when: cifmw_edpm_deploy_hci | default(false) | bool
+      ansible.builtin.meta: clear_facts
+
+- name: Deploy Ceph on EDPM nodes
+  vars:
+    storage_network_range: 172.18.0.0/24
+    storage_mgmt_network_range: 172.20.0.0/24
+  ansible.builtin.import_playbook: ceph.yml
+  when: cifmw_edpm_deploy_hci | default(false) | bool
+
+- name: Continue HCI deploy
+  hosts: "{{ cifmw_target_host | default('localhost') }}"
```

Comment:
sanity check is this working OK with respect to the facts you cleared above?

the clear_facts is targetted to 

```
hosts: edpm
```

but this one is cifmw_target_host
are they the same?

Reply:
Yes, this looks correct, it is the same as the first edpm_deploy run. These roles are going to run on 'localhost', and interact with OCP to deploy the manifiests. At this point, all manifests already have all the info about edpm-compute nodes (aka edpm/computes groups)
There are a few places only we run things directly on computes/edpm targets like in ceph.yml playbook. But most of the things are configured using DataPlane services or install_yamls targets

Reply:
thanks for checking marking it resolved